pkg/chunk/storage/factory.go

```diff
@@ -57,20 +59,18 @@ type StoreLimits interface {
 
 // Config chooses which storage client to use.
 type Config struct {
-	Engine                 string                  `yaml:"engine"`
-	AWSStorageConfig       aws.StorageConfig       `yaml:"aws"`
-	AzureStorageConfig     azure.BlobStorageConfig `yaml:"azure"`
-	GCPStorageConfig       gcp.Config              `yaml:"bigtable"`
-	GCSConfig              gcp.GCSConfig           `yaml:"gcs"`
-	CassandraStorageConfig cassandra.Config        `yaml:"cassandra"`
-	BoltDBConfig           local.BoltDBConfig      `yaml:"boltdb"`
-	FSConfig               local.FSConfig          `yaml:"filesystem"`
-
-	IndexCacheValidity time.Duration `yaml:"index_cache_validity"`
-
-	IndexQueriesCacheConfig cache.Config `yaml:"index_queries_cache_config"`
-
-	DeleteStoreConfig purger.DeleteStoreConfig `yaml:"delete_store"`
+	Engine                  string                   `yaml:"engine"`
+	AWSStorageConfig        aws.StorageConfig        `yaml:"aws"`
+	AzureStorageConfig      azure.BlobStorageConfig  `yaml:"azure"`
+	GCPStorageConfig        gcp.Config               `yaml:"bigtable"`
+	GCSConfig               gcp.GCSConfig            `yaml:"gcs"`
+	CassandraStorageConfig  cassandra.Config         `yaml:"cassandra"`
+	BoltDBConfig            local.BoltDBConfig       `yaml:"boltdb"`
+	FSConfig                local.FSConfig           `yaml:"filesystem"`
+	IndexCacheValidity      time.Duration            `yaml:"index_cache_validity"`
+	GrpcConfig              grpc.Config              `yaml:"grpc-store"`
```

Comment:
The convention is `_` in YAML, `-`  in CLI flags:

```suggestion
	GrpcConfig              grpc.Config              `yaml:"grpc_store"`
```